lib/matplotlib/axes/_axes.py

```diff
@@ -4588,13 +4588,17 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             s = (20 if mpl.rcParams['_internal.classic_mode'] else
                  mpl.rcParams['lines.markersize'] ** 2.0)
         s = np.ma.ravel(s)
+        scaling = 2
         if (len(s) not in (1, x.size) or
                 (not np.issubdtype(s.dtype, np.floating) and
                  not np.issubdtype(s.dtype, np.integer))):
             raise ValueError(
                 "s must be a scalar, "
                 "or float array-like with the same size as x and y")
 
+        if markersize is not None:
+            s = np.ma.ravel(markersize)
```

Comment:
what about squaring this, so it just cancels out the sqrt later? Then you wouldn't have to carry around the scaling.
```suggestion
            s = np.ma.ravel(markersize) ** 2
```

Do we always want to say that markersize takes precedence? Error/warn if both given?

Another option would be to have a `markerscaling` keyword argument that could be a string that controls whether to scale linearly or with the sqrt of the size.

Reply:
We'll for sure error out if both are given. I don't think it makes sense to give precedence to either. I just haven't implemented that because I just wanted to see if this idea works.

`markerscaling` is a good name. I like it.

Reply:
Wait, I think I misunderstood your `markerscaling` comment. Were you suggesting that we keep `s` as it is and just add `markerscaling` to control linear or quadratic?

Reply:
I see your comment on the linked issue now about not wanting to square the size due to the inconsistency between set/get_sizes then, that makes sense to me, so probably ignore my comment about squaring above.

Yes, my comment meant to make `markerscaling='linear'` take a string, which would then just change how `s=[5, 6, 7]` is interpreted, rather than having a user choose whether to put their data in `markersize` or `size`.  I guess another option would be to do something like `markerscaling=1`, and then `np.sqrt(np.power(s, markerscaling))` in the collection... Which is similar to what you have here already.

Reply:
One of the reasons for the issue and this PR is to have a more readable name for marker sizes too as `s` has no meaning by itself. I guess if `size` were allowed, this would not be that much of a problem but it's not. On the other hand, `plot` allows `markersize` to manage the sizes of the markers when not a line so having something similar here would also give more consistency in terms of usage. (I don't know how many times I've had to refer the documentation because I tried markersize with scatter first).